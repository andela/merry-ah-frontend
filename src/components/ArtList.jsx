import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import startCount from '../utils/startCount';
import counter from '../utils/counter';
import likeCount from '../assets/images/like.png';
import commentCount from '../assets/images/comment.png';
import { truncateString } from '../utils/truncateString';
/**
 * ArtList Component
 */
/**
   * @param {function} render
   *  @returns {JSX} jsx
*/
const ArtList = ({ user }) => (
     <div className='p-2 m-3 col-12'>

      <div className="row container" id="art-list">
        <p className="profile-header col-12 mt-5">My Works</p>
        { !user.isGettingArts && user.userArts.map((art, i) => (
            <div key={i} className="col-md-3 col-sm-12">
              <div
                className="my-card"
                style={{
                  backgroundImage: `url(${art.featuredImg})`,
                  backgroundColor: 'rgba(0, 0, 0, 0.5)',
                }}
              >
                <div className="card-category">{art.Category.categoryName}</div>
                <div className="card-container">
                  <div className="card-title">
                    <h6>
                      {' '}
                      <Link to={`/arts/${art.slug}`}> {art.title}</Link>{' '}
                    </h6>
                    <p>
                    {truncateString(
                      art.description.replace(/<p>/g, '').replace(/<\/p>/g, ''), 40,
                    )}
                    </p>
                  </div>
                </div>
              </div>
              <div className="card-footer">
                <div className="comments art-summary">
                  <p>
                    <img src={likeCount}
                    alt="" /> {counter(art.LikesCount)}{' '}
                  </p>
                  <p>
                    <img src={commentCount}
                    alt="" /> {counter(art.CommentsCount)}{' '}
                  </p>
                </div>
                <div className="ratings">
                {startCount(art.CalculatedRate)}
                </div>
              </div>
            </div>
        ))}
        </div>

        </div>
);

ArtList.propTypes = {
  user: PropTypes.shape({
    userArts: PropTypes.array,
    isGettingArts: PropTypes.bool.isRequired,
  }),
};
export default ArtList;
