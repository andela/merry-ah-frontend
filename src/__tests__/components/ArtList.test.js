/* eslint-disable import/no-unresolved */
import 'babel-polyfill';
import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import ArtList from '../../components/ArtList.jsx';
import { props } from '../__mocks__/usersMock';

let wrapper;
beforeEach(() => {
  wrapper = mount(
    <MemoryRouter>
      <ArtList {...props} />
    </MemoryRouter>,
  );
});
describe('ArtList Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
