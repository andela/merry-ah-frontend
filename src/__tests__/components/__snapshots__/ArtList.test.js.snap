// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtList Component should match snapshot 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "iyy72o",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "iyy72o",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ArtList
      auth={
        Object {
          "user": Object {
            "id": 7,
          },
        }
      }
      editProfile={[Function]}
      getUserArts={[Function]}
      getUserProfile={[Function]}
      user={
        Object {
          "isGettingArts": false,
          "userArts": Array [
            Object {
              "Category": Object {
                "categoryName": "Poetry",
              },
              "description": "some descriptions",
              "featuredImg": "image.url",
              "slug": "isuguixu",
              "title": "some title",
            },
          ],
          "userProfile": Object {
            "id": 7,
            "profile": Object {
              "bio": "my bio",
              "firstName": "Juliet",
              "imgURL": "img.URL",
              "lastName": "Ezekwe",
            },
            "userType": "artist",
          },
        }
      }
    >
      <div
        className="p-2 m-3 col-12"
      >
        <div
          className="row container"
          id="art-list"
        >
          <p
            className="profile-header col-12 mt-5"
          >
            My Works
          </p>
          <div
            className="col-md-3 col-sm-12"
            key="0"
          >
            <div
              className="my-card"
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.5)",
                  "backgroundImage": "url(image.url)",
                }
              }
            >
              <div
                className="card-category"
              >
                Poetry
              </div>
              <div
                className="card-container"
              >
                <div
                  className="card-title"
                >
                  <h6>
                     
                    <Link
                      replace={false}
                      to="/arts/isuguixu"
                    >
                      <a
                        href="/arts/isuguixu"
                        onClick={[Function]}
                      >
                         
                        some title
                      </a>
                    </Link>
                     
                  </h6>
                  <p>
                    some descriptions
                  </p>
                </div>
              </div>
            </div>
            <div
              className="card-footer"
            >
              <div
                className="comments"
              >
                <p>
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                   
                   
                </p>
                <p>
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                   
                   
                </p>
              </div>
              <div
                className="ratings"
              >
                <img
                  alt=""
                  src="test-file-stub"
                />
                <img
                  alt=""
                  src="test-file-stub"
                />
                <img
                  alt=""
                  src="test-file-stub"
                />
                <img
                  alt=""
                  src="test-file-stub"
                />
                <img
                  alt=""
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </ArtList>
  </Router>
</MemoryRouter>
`;
